ui/src/common/actions.ts

```diff
@@ -664,9 +674,46 @@ export const StateActions = {
         trackList.push(x);
       });
     };
+    let trackLikeSrc: TrackState | TrackGroupState = state.tracks[args.srcId];
+    let trackLikeDst: TrackState | TrackGroupState = state.tracks[args.dstId];
+    let srcParent: string | undefined;
+    let dstParent: string | undefined;
+    if (trackLikeSrc && trackLikeSrc.name) {
+      srcParent = trackLikeSrc.trackGroup;
+    } else {
+      trackLikeSrc = state.trackGroups[args.srcId];
+      srcParent = trackLikeSrc.parentGroup;
+    }
+
+    if (trackLikeDst && trackLikeDst.name) {
+      dstParent = trackLikeDst.trackGroup;
+    } else {
+      trackLikeDst = state.trackGroups[args.dstId];
+      dstParent = trackLikeDst.parentGroup;
+    }
 
-    moveWithinTrackList(state.pinnedTracks);
-    moveWithinTrackList(state.scrollingTracks);
+    if (srcParent && srcParent === dstParent) {
+      if (srcParent === SCROLLING_TRACK_GROUP) {
+        moveWithinTrackList(state.scrollingTracks);
+      } else {
+        moveWithinTrackList(state.trackGroups[srcParent].sortOrder);
+      }
+    }
+    // Currently we don't support making track groups pinned
+    if ('trackGroup' in trackLikeSrc) {
+      moveWithinTrackList(state.pinnedTracks);
+    }
+  },
+
+  moveIdlesToBottom(state: StateDraft, _: {}): void {
+    for (const group of Object.values(state.trackGroups)) {
+      if (group.name.search(/\bIdle\b/) >= 0) {
```

Comment:
I would rather have a list of group IDs passed in to the `moveIdlesToBottom` action in the argument, so that this action doesn't have to be concerned with naming conventions like this. The caller should determine which groups to move. Then this action could be named `sortTrackGroupsToBottom`.